packages/core/src/sync-historical/index.ts

```diff
@@ -321,14 +326,17 @@ export const createHistoricalSync = async (
       }
     }
 
-    // Add `transactionHashes` to the sync-store.
+    // Add `transactionHashes` to `transactionCache`.
     if (transactionHashes !== undefined) {
-      await args.syncStore.insertTransactions({
-        transactions: block.transactions.filter((transaction) =>
-          transactionHashes.has(transaction.hash),
-        ),
-        chainId: args.network.chainId,
-      });
+      // Filter out transactions that aren't relevant
+      const newTransactions = block.transactions.filter(({ hash }) =>
+        transactionHashes.has(hash),
+      );
+
+      // Add `newTransactions` to the cache
+      for (const transaction of newTransactions) {
+        transactionsCache.add(transaction.hash);
+      }
```

Comment:
I don't think the `block.transactions.filter( ... )` is necessary here? Could do
```
// Add `newTransactions` to the cache
if (transactionHashes !== undefined) {
  for (const transactionHash of transactionHashes) {
    transactionsCache.add(transaction.hash);
  }
}
```

Someday we could use [`Set.prototype.union()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/union)!

Reply:
Nice catch - after looking at this more closely it made sense to just include the transaction specific logic in the filter functions. Now `syncLogFilter` and `syncCallTraceFilter` manage adding the required transaction to the `transactionCache`